components/constellation/webview.rs

```diff
@@ -17,6 +17,12 @@ pub struct WebViewManager<WebView> {
 
     /// Whether the latest webview in focus order is currently focused.
     is_focused: bool,
+
+    /// The web view that would be visible in a containing native window.
+    visible_webviews: HashSet<TopLevelBrowsingContextId>,
+
+    /// Whether our native window is visible, or true if there is no such window.
+    native_window_is_visible: bool,
 }
```

Comment:
I would like to know if we could simplify it with the set of `visible_webviews` only. This is also related to the comment above about `WebViewVisibilityChanged`. Could we keep only either `WebViewVisibilityChanged` or `ShowWebView`/`HideWebView` variant?

Because users might not understand internal parts like constellation and compositor, when I see the "visibility" configuration, I would expect it to be a boolean option to toggle as well. Could compositor and constellation share only one visibility set and reflect based on it?

Reply:
After some discussion, we decide to replace with `shown_webviews` and `invisible_webviews`. We'll think about how to simplify it once multi windows support is here.